harness/determined/common/api/authentication.py

```diff
@@ -107,6 +120,10 @@ def _init_session(
         if token is not None:
             return UsernameTokenPair(session_user, token)
 
+        # If session_user has no valid token in neither the token store nor the environment, then
+        # we prompt for a password for session_user under the following conditions:
+        # * session_user is NOT 'determined'.
+        # * session_uses is 'determined' but does not have the default determined password.
```

Comment:
We'll need to create a new token, so we'll need a password.  If there was no requested user and we ended up falling back to the default username `determined`, then we can fall back to the default login as well.  Otherwise ask the user for their password.